proxy/tcp/tls_clienthello_test.go

```diff
@@ -0,0 +1,57 @@
+package tcp
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/fabiolb/fabio/assert"
+)
+
+func TestReadServerNameBadData(t *testing.T) {
```

Comment:
Can you check whether you can also write a table driven test like the one I've outlined above?